### PR TITLE
feat: start with data default option added

### PR DESCRIPTION
**Pull request by sneha122**
## Description

We recently delivered an experiment for onboarding revamp, where 50% of the new users would see onboarding flow (with 3 options of start with data, start from scratch and start from templates), this experiment proved to be useful and as a next phase of that, we decided to experiment with providing start with data option as default, so when a new user signs up, they will see start with data option by default. This PR addresses that experiment.

Notion doc: https://www.notion.so/Productize-Onboarding-revamp-c67bdce04fcb451b95690d2ea2ac60c3

#### PR fixes following issue(s)
Fixes # (issue number)
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Introduced a new feature flag to control the default behavior for starting applications with data.
- **Enhancements**
	- Updated the application creation flow to conditionally render options based on the new feature flag.
- **UI/UX**
	- Added styling to center align and adjust padding for a component in the Schema View mode.

### CODE REVIEW

File: app/client/src/ce/pages/Applications/CreateNewAppsOption.tsx [hunk 1]
```diff
@@ -409,6 +447,10 @@ const CreateNewAppsOption = ({
         })),
       );
 
+    if (isEnabledForStartWithDataDefault) {
+      onClickStartWithData();
```
Comment (AmanAgarwal041) [inline app/client/src/ce/pages/Applications/CreateNewAppsOption.tsx:451]: Shouldn't we add some other event ? Just confirming

Comment (sneha122) [reply app/client/src/ce/pages/Applications/CreateNewAppsOption.tsx:451]: Um I am not sure about that but with this code for default option, start with data would be triggered, may be we can include the flag as well in that event to identify whether it was default flow or normal 3 option flow. I will confirm this with Balaji. Thanks Aman!!

**Review by ayushpahwa — DISMISSED** [review]

**Review by ayushpahwa — APPROVED** [review]

**Pull request merged**